crypto/bn/bn_rsa_fips186_4.c

```diff
@@ -235,6 +259,27 @@ int bn_rsa_fips186_4_derive_prime(BIGNUM *Y, BIGNUM *X, const BIGNUM *Xin,
     if (Xin != NULL && BN_copy(X, Xin) == NULL)
         goto err;
 
+    /*
+     * We need to generate a random number X in the range
+     * 1/sqrt(2) * 2^(nlen/2) <= X < 2^(nlen/2).
+     * We can rewrite that as:
+     * base = 1/sqrt(2) * 2^(nlen/2)
+     * range = ((2^(nlen/2))) - (1/sqrt(2) * 2^(nlen/2))
+     * X = base + random(range)
+     * We only have the first 256 bit of 1/sqrt(2)
+     */
+    if (Xin == NULL) {
+        if (bits < BN_num_bits(&bn_inv_sqrt_2))
+            goto err;
+        if (BN_copy(base, &bn_inv_sqrt_2) == NULL
+            || !BN_add(base, base, BN_value_one())
```

Comment:
Alternatively, drop one bit at the end of `bn_inv_sqrt_2` to add the leading `1` bit and shift one more.

Reply:
So you're suggesting to change the number from:
0xB504F333F9DE6484597D89B3754ABE9F1D6F60BA893BA84CED17AC8583339915
to:
0xB504F333F9DE6484597D89B3754ABE9F1D6F60BA893BA84CED17AC8583339916
(or 0x83339915UL to 0x83339916UL), to avoid BN_add(), which is easy to do. But I don't get your comment about one more limb.

Removing a bit doesn't make sense. You would get this instead:
0x5A827999FCEF32422CBEC4D9BAA55F4F8EB7B05D449DD426768BD642C199CC8A
You still need to add 1, and then shift 1 more bit.

Reply:
You're correct about the extra limb, just adding the one as you've done is good.